### ISSUE TITLE
Time to update Thor 

### ISSUE THREAD

**Issue by freibuis** [castwide/solargraph#254]
currently, solargraph relies on thor
```ruby
s.add_runtime_dependency 'thor', '~> 0.19', '>= 0.19.4'
```
I think it is time to upgrade as some core frameworks like rails are forcing thor to a min of thor version 1 . This might cause some issues with beginners who do not know how to fix dependencies 

0.19.4 is getting pretty old now

```
VERSIONS:
1.0.1 - December 17, 2019 (55 KB)
1.0.0 - December 13, 2019 (55 KB)
0.20.3 - November 10, 2018 (54.5 KB)
0.20.2 - November 10, 2018 (54.5 KB)
```

has anyone done any tests with the new version of thor?

**Comment by castwide** [comment]
I'm tempted to put this change on the back burner for the moment, because the update to 1.0 seems totally backwards compatible without being immediately necessary. Please let me know if I'm wrong, or if anything breaks that I didn't expect. Either way, this change is officially added to the backlog.

**Comment by castwide** [comment]
On second thought, we might as well proceed with this update sooner than later. The dependency to thor is ~> 1.0 in the master branch and the change will be included in the next release. Tests look fine in Travis.

**Comment by freibuis** [comment]
awesome. I am testing it out now.  no issues so far

**Comment by freibuis** [comment]
nope. It is breaking for me..  this might be a me issue tho. moved to ruby 2.7.0 

I cant tell if yard is breaking because of thor or ruby 2.7.0  is broke.

I will spin up 2.6.5/2.6.4 and retest

**Comment by castwide** [comment]
It might be YARD and Ruby 2.7: https://github.com/lsegal/yard/pull/1296

**Comment by freibuis** [comment]
yeah 
```
(e2mmap gem) is no longer bundled gems.
```
definitely see the complaints about this missing `e2mmap` file
2.6.5 works fine..  
2.7.0 is definitely the issue
so ruby 2.7.0 moved a lot of internals to external gems I am sure there will be more to come.

so back to 2.6.5 until yard is fixedfor me.

Should the yard project be watched and bump the min yard dep once it is has 2.7.0 support?

**Comment by castwide** [comment]
Yeah, I plan to bump the yard dependency as soon as possible. Besides the issue with Ruby 2.7, there's another update that will improve API coverage in the core documentation.

**Comment by freibuis** [comment]
looks like yard has been upgraded now and now works for me :)
all back to normal.. just some 2.7.0 warnings tho :+1:

**Comment by castwide** [comment]
Thor update released in gem 0.38.1.
